protobuf/module_msg.proto

```diff
@@ -206,6 +206,26 @@ message DRRArg {
   uint32 num_flows = 1;  /// Number of flows to handle in module
   uint64 quantum = 2;  /// the number of bytes to allocate to each on every round
   uint32 max_flow_queue_size = 3; /// the max size that any Flows queue can get
+  
+  enum QueueType {
+    LLRING = 0;
+    CODEL = 1;
+  }
+  
+  message Llring {
+   uint32 size  = 1; /// the initial size of each flow's queue 
+   bool multiple_producer = 2; /// True if only allow multples producers for the Llring 
+   bool multiple_consumer = 3; /// True if only allow multiple consumers for Llring
+  }
+
+  message Codel {
+    uint64 target = 1; /// the desired delay in microseconds that codel tries to maintain
+    uint64 window = 2; /// window in microseconds between changing states in codel 
+  }
+
+  QueueType type = 4; /// which queuetype is used for each flow
```

Comment:
Instead of having `QueueType`, we can do
```
oneof params {
  Llring ring = 4;
  Codel codel = 5;
}
```

This is more semantically correct (only either of them can be set), and simpler (neither `Enum` nor type indicator is required).

In my opinion, `Llring ring` should be renamed `DropTailQueue` or something like that. The fact that the underlying queue is lockless is just internal implementation details; external users need to know what it does (plain old queue without active queue management), not how it is implemented.

I don't see much value for `multiple_producer` and `multiple_consumer` parameters. These parameters should be hidden from users; the system must not crash even if some parameters are not correctly set. Also, given that the module is not thread-safe, there is no point for mult-producer/consumer parameters set.